### ISSUE TITLE
Didn't work as expected

### ISSUE THREAD

**Issue by quantumleap33** [Hube2/wpforms-ajax-submit#3]
Hi there. I entered the class under the **Settings->General->Form CSS Class** option. When submited the form, the ajax overlay covered the whole form and after submission even though the URL remained the same, only the confirmation message was displayed on screen and not the form, essentially functioning as previously in a non ajax way. Tested on latest version of Firefox.

**Comment by Hube2** [comment]
The only change that happens is that the form is submitted through AJAX, other than this the operation of WPForms is not altered. Normally, when you submit a form the confirmation is shown without the form when the page loads, unless you have set up the form to redirect to another page/url after submission and this would not be changed either. WPForms does not re-display the form that was just submitted during normal operation either.

Exactly what were you expecting to happen? If the form is submitted successfully shouldn't the confirmation message be shown?

**Comment by quantumleap33** [comment]
Yes,  it should but with the previous form present as well, at least in my opinion. Like Contact Form 7 is doing it where after hitting the Submit buttom, you get the confirmation underneath the button and the form's fields get emptied but still present. Otherwise, I don't see any difference with how your plugin is managing it and the dafault WP Forms Lite behaviour apart from the URL staying the same. The underlined ajax effect is not visible to the user, if you know what I mean. User still thinks he/she got redirected to a different page, there is a disruption on the UI flow.

**Comment by Hube2** [comment]
My goal is not to alter anything about the way that WP forms functions other that to submit the form using AJAX. WPForms does not re-display the form. Both the form and the thank you message are shown using the same shortcode. WPForms remembers that the user just submitted the form and does not re-display it.

The main reason that I created this was in order to use the form in a modal window. Why, because without it, if the form is in a modal window, reloading of the page causes the modal window to close. WPForms suggested using a paid service to use modals that remained open and for me that is an unacceptable solution.

AJAX submission has other benefits as well, for example, reducing the time it takes to submit a form. 

My goal was to implement AJAX for WPForms without affecting the user experience provided by WPForms in any way. If WPForms themselves implemented an AJAX submission process, my guess is that it would work the same way that I have implemented it. I understand how CF7 does things, but the goal was for me is not to mimic the way another plugin does it. 

In my opinion, the change that you are requesting should be on the WPForms end. Perhaps a setting in the form to re-display the form with the confirmation. 

Although I don't see the purpose to having the form re-displayed. Do you want to allow the form to be submitted multiple times? If this is the case then you could include an link to the current page in the confirmation message back to the current page with text along the lines of "Submit this form again", this would reload the page including the form.

The main problem here is that WPForms replaces all of the content in there container (IE: the form) with the confirmation message. It only returns this content with no indication on if the form submitted successfully or not. To do this I would need to somehow cache the displayed form on the client side. Then I would need to somehow detect if the content returned included the form or not and if the form submission was successful or not and then also add the form back into the container when needed.

I am in the process of improving this plugin so that it allows developers to alter the overlay color and spinner image as well as give more control over how it is displayed. If I have time I will look into whether or not the above is possible and consider the possibility of adding it. At the moment I feel that this is out of the scope of the intended purpose of this add on.
